numpy/core/src/multiarray/compiled_base.c

```diff
@@ -1783,28 +1792,30 @@ unpack_bits(PyObject *input, int axis)
         unpack_init = 1;
     }
 
-    NPY_BEGIN_THREADS_THRESHOLDED(PyArray_DIM(new, axis));
-
-    n_in = PyArray_DIM(new, axis);
+    in_n = count / 8;
+    in_tail = count % 8;
     in_stride = PyArray_STRIDE(new, axis);
     out_stride = PyArray_STRIDE(out, axis);
 
-    while (PyArray_ITER_NOTDONE(it)) {
+    NPY_BEGIN_THREADS_THRESHOLDED(in_n);
```

Comment:
Not really sure how this macro works to be honest. Is the proper way to do this as shown, or is it better to do `NPY_BEGIN_THREADS_THRESHOLDED(in_n + (in_tail != 0));`

Reply:
Given how the loops worked out, I expect this will be fine though.

Reply:
It tests whether the argument is above a threshold (probably a few hundred), under which we assume that threading overhead is too large to be worth the trouble

Reply:
I see. The +/- 1 is indeed immaterial then.

Reply:
I think `PyArray_Size(in) / size_along_axis * unpacked size` might be a more honest option, but my guess is that in most cases it won't matter. I'll make the change anyway.